src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -593,7 +593,12 @@ public bool IsDotNetFramework
 
         internal static ManagedPlatform ComputeIsDotNetCore(MetadataReader metadataReader)
         {
-            if (metadataReader.AssemblyReferences.Count == 0)
+            if (metadataReader?.AssemblyReferences == null)

```

Comment:
>metadataReader [](http://example.com/codeflow?start=16&length=14)

why is this value null? it looks like this method is protected universally by an IsManaged check which makes me think this should be non-null. but you are seeing a condition when this isn't so?

Reply:
The question is, if we have a managed binary that raises the exception you have provoked, is it impossible for us to have a non-null metadata reader.

Is that the case? If it is, then your changes are reasonable. But if what's happening here is that we're simply not processing the exceptions properly to finish instantiating a valid metadata reader, that's the change that we want.